source/FreeRTOS_IP_Utils.c

```diff
@@ -1764,3 +1773,131 @@ uint16_t usChar2u16( const uint8_t * pucPtr )
              ( ( ( uint32_t ) pucPtr[ 1 ] ) ) );
 }
 /*-----------------------------------------------------------*/
+
+/**
+ * @brief Every IPv6 end-point has a solicited node multicast address and a corresponding
+ * multicast MAC address. The IPv6 solicited node address also has an MLD reports associated
+ * with it. This functions manages both the MAC address and the MLD report associated with
+ * the end-point's solicited node address. On network UP, this function registers the MAC address
+ * with the network driver's filter and creates and MLD report for the UPv6 multicast address.
+ * On network DOWN, the function unregisters the MAC address and removes the MLD report.
+ * This is a "convenience" function that keeps all these tasks under one roof for easier maintenance.
+ *
+ * @param[in] pxEndPoint The end-point for which a network up/down event is being handled.
+ * @param[in] xNetworkGoingUp pdTRUE when the network goes UP, pdFALSE when the network goes DOWN.
+ */
+#if ( ipconfigIS_ENABLED( ipconfigUSE_IPv6 ) )
+    void vManageSolicitedNodeAddress( struct xNetworkEndPoint * pxEndPoint,
+                                      BaseType_t xNetworkGoingUp )
+    {
+        /* _EP_: I did my best to verify that both pxEndPoint and pxEndPoint->pxNetworkInterface cannot be
+         * NULL but I'm leaving the checks below until I get a second opinion. */
+        configASSERT( pxEndPoint != NULL );
+        configASSERT( pxEndPoint->pxNetworkInterface != NULL );
```

Comment:
I think we can leave the asserts in the code. Anyways in production code the ASSERTS are supposed to be compiled out so should not be a problem 🙂

Reply:
+1
As ` vManageSolicitedNodeAddress()` is only called internally, this `configASSERT()` is a useful and safe test for as long as we are testing.